libraries/DoublyLinkedList/DLLNode.h

```diff
@@ -0,0 +1,27 @@
+//  Created by Andrew Florjancic on 9/18/24.
+//  Copyright © 2024 Andrew Florjancic. All rights reserved
+#ifndef DLLNode_h
+#define DLLNode_h
+
+// A Generic Doubly Linked List Node.
+template <class T>
+class DLLNode {
+    public:
+        // The data stored in the node.
+        const T data;
+
+        // A Pointer to the previous node in the list.
+        DLLNode<T>* previous_node = nullptr;
```

Comment:
These probably shouldn't be public. The Menu does need to traverse the list to display items on the LCD but doesn't actually need to modify the previous_node or next_node pointers. Let's just make these private and include some get methods.

Reply:
The DoublyLinkedList needs to modify the pointers to previous and next nodes. I'm fine with just leaving these public.